packages/amplify-util-uibuilder/src/event-handlers/handle-PrePush.ts

```diff
@@ -0,0 +1,15 @@
+import { $TSContext } from 'amplify-cli-core';
+import { printer } from 'amplify-prompts';
+import { prePushHandler } from '../utils/prePushHandler';
+
+/**
+ * The code to run before a push.
+ */
+export const run = async (context: $TSContext): Promise<void> => {
+  try {
+    await prePushHandler(context);
+  } catch (e) {
+    // Swallow all errors from the uibuilder plugin
+    printer.debug(e);
+  }
```

Comment:
Along the same lines as my previous comment, there seems to be a lot of duplicated code here and it seems like extensibility would get exponentially more difficult for each event. There's probably an opportunity to dry up some of this code.

Reply:
While I agree with the sentiment, I don't think refactoring all of the event handlers is in-scope for this PR. It's a significant enough refactor to warrant it's own PR.